sql/service/src/main/scala/org/apache/spark/sql/service/cli/session/SessionManager.scala

```diff
@@ -0,0 +1,346 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.service.cli.session
+
+import java.io.{File, IOException}
+import java.util
+import java.util.Date
+import java.util.concurrent._
+
+import scala.collection.JavaConverters._
+
+import org.apache.commons.io.FileUtils
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.service.{CompositeService, SparkThriftServer2}
+import org.apache.spark.sql.service.cli.{ServiceSQLException, SessionHandle}
+import org.apache.spark.sql.service.cli.operation.OperationManager
+import org.apache.spark.sql.service.internal.ServiceConf
+import org.apache.spark.sql.service.rpc.thrift.TProtocolVersion
+import org.apache.spark.sql.service.server.{SparkServer2, ThreadFactoryWithName}
+
+class SessionManager(sparkServer2: SparkServer2, sqlContext: SQLContext)
+  extends CompositeService(classOf[SessionManager].getSimpleName)
+    with Logging {
+
+  private var sqlConf: SQLConf = null
+  private val handleToSession = new ConcurrentHashMap[SessionHandle, ServiceSession]
+  private val operationManager = new OperationManager
+  private var backgroundOperationPool: ThreadPoolExecutor = null
+  private var isOperationLogEnabled: Boolean = false
+  private var operationLogRootDir: File = null
+
+  private var checkInterval: Long = 0L
+  private var sessionTimeout: Long = 0L
+  private var checkOperation: Boolean = false
+
+  private var shutdown: Boolean = false
+  // The SparkServer2 instance running this service
+
+  override def init(sqlConf: SQLConf): Unit = {
+    this.sqlConf = sqlConf
+    // Create operation log root directory, if operation logging is enabled
+    if (sqlConf.getConf(ServiceConf.THRIFTSERVER_LOGGING_OPERATION_ENABLE).asInstanceOf[Boolean]) {
+      initOperationLogRootDir()
+    }
+    createBackgroundOperationPool()
+    addService(operationManager)
+    super.init(sqlConf)
+  }
+
+  private def createBackgroundOperationPool(): Unit = {
+    val poolSize = sqlConf.getConf(ServiceConf.THRIFTSERVER_ASYNC_EXEC_THREADS)
+    logInfo("SparkServer2: Background operation thread pool size: " + poolSize)
```

Comment:
offtopic comment: Why SparkServer2? Just because it was HiveServer2? Maybe make it just SparkSQLServer?

Reply:
> offtopic comment: Why SparkServer2? Just because it was HiveServer2? Maybe make it just SparkSQLServer?

I have make the change in my local, since want one pr do one thing, and these change have conflict. We are waiting for your advise since we want simplify the code.